### ISSUE TITLE
require signature is incorrect

### ISSUE THREAD

**Issue by Anahkiasen** [s-panferov/awesome-typescript-loader#35]
The current `require` signature is the following:

```ts
interface WebpackRequire {
    (id: string): any;
    ensure(ids: string[], callback: WebpackRequireEnsureCallback): void;
}
```

But this doesn't support AMD imports as [documented here](http://webpack.github.io/docs/code-splitting.html#require).

**Comment by Anahkiasen** [comment]
Also needs support for the optional third parameter for [named chunks](http://webpack.github.io/docs/code-splitting.html#named-chunks):

```ts
require.ensure([], function() { }, 'name-of-chunk');
```

**Comment by s-panferov** [comment]
@Anahkiasen 	thanks! If you want a workaround, rust use `emitRequireType=false` loader option and define this functions yourself in one of your .d.ts files. I'll try to fix it tomorrow morning (GMT+3)

**Comment by s-panferov** [comment]
@Anahkiasen, I added the signature https://github.com/s-panferov/awesome-typescript-loader/commit/970ff87595505f1cda435ee0cb5283d79fa34fc2.

Landed as v0.8.0. Please free to reopen if something is wrong. Thanks again for the report.
